src/compile/data/aggregate.ts

```diff
@@ -49,9 +48,21 @@ export class AggregateNode extends DataFlowNode implements NewFieldNode, Depende
   /** dictionary mapping field name => dict set of aggregation functions */
   private measures: Dict<StringSet>;
 
+  public clone(): this {
+    const cloneObj = new (<any>this.constructor);
```

Comment:
This looks like a hack to get around the fact that `AggregateNode()` constructor requires `model` as input. 

I wonder if we should make AggregateNode takes `dimensions` and `measures` as input and make a factory that takes a model as input?  

I believe making aggregateNode taking `dimensions` and `measures` as input might make writing test easier as well (not sure about this latter comment though).